### ISSUE TITLE
1px Black Bars on edges

### ISSUE THREAD

**Issue by dragons4life** [mpv-player/mpv#3283]
Windows 10 / Latest BitBucket build

<video src="https://a.uguu.se/x2lcrMkKvXgz_802143015b.webm"></video>

**Comment by wiiaboo** [comment]
mpv.conf would be useful too.

**Comment by dragons4life** [comment]
http://pastebin.com/raw/USSqMXyS

**Comment by haasn** [comment]
1. The issue template isn't there to be ignored. A log is helpful.

2. That's a lot of configuration options. Try reproducing the issue with `-no-config -no-border`.

**Comment by dragons4life** [comment]
--no-config --no-border
https://a.pomf.cat/ijblak.webm

**Comment by rossy** [comment]
This bugs me too. I think it's due to different rounding in the aspect calculation between video/out/aspect.c, video/out/win_state.c and the WM_SIZING handler in w32_common.c. Maybe it would be an idea to reduce the amount of code responsible for fitting video to the correct aspect ratio?

Alternatively, I've been playing around with a patch that uses a 1px fudge factor for when the video size almost matches up to the window size, but this might not be the right solution, especially since I think I noticed some 2px black borders in your videos: rossy/mpv@a116feb

**Comment by dragons4life** [comment]
So.. where any of able to reproduce the same bug?

**Comment by rossy** [comment]
Yeah, if it's what I'm thinking of, I get this all the time. Here's a 1px border on both the left and right sides:
![image](https://cloud.githubusercontent.com/assets/162837/16360785/762995b8-3bb7-11e6-85cc-1b35ba948597.png)

**Comment by dragons4life** [comment]
Yeah, at first I though it was the videos, so I didn't care that much, eventually I knew it wasn't but I never really got to reporting it.

**Comment by nagisa** [comment]
Can reproduce on linux/x11 with vo=vaapi as well.

**Comment by wm4** [comment]
Try ``--panscan=1``.
